common/common_test.go

```diff
@@ -856,3 +856,36 @@ func TestErrorCollector(t *testing.T) {
 	require.True(t, ok, "Must return a multiError")
 	assert.Len(t, errs.Unwrap(), 2, "Should have 2 errors")
 }
+
+// TestBatch ensures the Batch function does not regress into common behavioural faults if implementation changes
+func TestBatch(t *testing.T) {
+	s := []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+	b := Batch(s, 3)
+	require.Len(t, b, 4)
+	assert.Len(t, b[0], 3)
+	assert.Len(t, b[3], 1)
+
+	b[0][0] = 42
+	assert.Equal(t, 1, s[0], "Changing the batches must not change the source")
+
+	require.NotPanics(t, func() { Batch(s, -1) }, "Must not panic on negative batch size")
+	done := make(chan any, 1)
+	go func() { done <- Batch(s, 0) }()
+	require.Eventually(t, func() bool { return len(done) > 0 }, time.Second, time.Millisecond, "Batch 0 must not hang")
```

Comment:
Yes. Previously it would hang. This is a regression test.

See L860
```
// TestBatch ensures the Batch function does not regress into common behavioural faults if implementation changes
```

I struggled slightly because I was considering saying "Regression test" in 874 where you're asking this, but it felt off because no merge of the failure has happened. We're protecting against a regression that's within the scope of the PR.

Reply:
I complain if there's no commentary, but I also complain when I don't _read_ commentary. There is no winning for you